### PR TITLE
Fixes #2212. Editor Scenario: Crashes on Exit.

### LINKED ISSUE #2212: Editor Scenario: Crashes on Exit
lNOeRpZ 1

My VS2022 install has become hooked and I can't debug right now. Sigh.

### PR DESCRIPTION
**Pull request by BDisp**
Fixes #2212 - The `historyText.Clear` must be called to reset `IsDirty` to false. The `LoadStream` method now also handle null terminated for the cases where they must be part of the text or not.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Views/TextView.cs [hunk 1]
```diff
@@ -1794,6 +1799,14 @@ void WrapTextModel ()
 		/// </summary>
 		public ContextMenu ContextMenu { get; private set; }
 
+		/// <summary>
+		/// Gets or sets if null terminated in a buffer array is considered as part of the <see cref="Text"/>.
+		/// </summary>
+		public bool AllowsNullTerminated {
```
Comment (tig) [inline Terminal.Gui/Views/TextView.cs:1805]: What is the user-scenario where a developer would set this property to new?

It seems like very obscure edge-case?

**Review by tig — CHANGES_REQUESTED** [review]

File: Terminal.Gui/Views/TextView.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs:1805]: See the unit test added. If it's encoded as `Unicode` it add a null terminated in the array byte.

File: UnitTests/TextViewTests.cs [hunk 2]
```diff
@@ -1930,6 +1929,42 @@ public void LoadStream_LF ()
 			Assert.Equal ($"This is the first line.{Environment.NewLine}This is the second line.{Environment.NewLine}", tv.Text);
 		}
 
+		[Fact]
+		public void LoadStream_IsDirty_AllowsNullTerminator ()
+		{
+			var text = "Testing";
```
Comment (tig) [inline UnitTests/TextViewTests.cs:1935]: I don't see any tests that show what happens if a `null` is in the middle of the text. Should there be a tests for that?

Comment (BDisp) [reply UnitTests/TextViewTests.cs:1935]: Yes you are right. I'll do it.

File: Terminal.Gui/Views/TextView.cs [hunk 3]
```diff
@@ -142,6 +142,11 @@ public override string ToString ()
 		/// </summary>
 		public int Count => lines.Count;
 
+		/// <summary>
+		/// Gets or sets if null terminated in a buffer array is considered as part of the <see cref="lines"/>.
+		/// </summary>
+		public bool AllowsNullTerminated { get; set; }
```
Comment (tig) [inline Terminal.Gui/Views/TextView.cs:148]: It doesn't matter if the `\0` is at the end of the line or not, right? So this has nothing to do with 'NullTerminated', but just `Null`.

Also, I believe the logic you implemented means that if this member is `true` any `\0` characters in the input stream will be preserved (and thus displayed in the editor). If `false` they will be ignored.

Thus, I believe this member should be named `AllowsNulls` instead. 

If I'm correct please rename it to `IgnoreNulls` and update the help text to something like 

> 'Gets or sets if '\0' characters will be preserved in the input stream by the <see cref="LoadStream(Stream stream)"/> method. If set to <see langref="false"/> (the default) any `\0` characters in the input stream will be preserved (and rendered in the TextView). If set to <see langref="true"/> `\0` characters will be stripped from the input stream."

And what will the editor show when `\0` characters are preserved in the input stream? Will it show a U+2400 (␀)?

I may be completely misunderstanding this though, so please correct me if I'm wrong!

**Review by tig — CHANGES_REQUESTED** [review]
See my comments about `AllowsNullTerminated`

File: Terminal.Gui/Views/TextView.cs (hunk 3 above)
Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs:148]: @tig I really I'm confused about this. This issue come up because I was getting wrong value on the unit test because of this statement:
```cs
var text = "Testing";
byte [] buff = System.Text.Encoding.Unicode.GetBytes (text);
```
`buff` returned an array length of 14, the double of 7 that is the length if the `text` variable. So, the `GetBytes` method was returning a null terminated for each character and not only at the end of the line. Maybe was my mistake of the using of the `GetBytes` method. If you think it's better not implementing this property at all I have no problem by remove it.
If you want to preserve I'm now with a doubt if I should named `AllowsNulls` or `IgnoreNulls`.

> And what will the editor show when \0 characters are preserved in the input stream? Will it show a U+2400 (␀)?

Yes it will.

Comment (tig) [reply Terminal.Gui/Views/TextView.cs:148]: I think you should simplify this fix to fix the crash in the most simple way possible.

Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs:148]: You are meaning I should discard this `AllowsNullTerminated` or whatever property, right?

Comment (BDisp) [reply Terminal.Gui/Views/TextView.cs]: Done @tig. I removed the `AllowsNullTerminated` property then.

**Review by tig — APPROVED** [review]
